crates/ascii-streamer/src/lib.rs

```diff
@@ -0,0 +1,75 @@
+#[macro_use]
+extern crate tracing;
+
+use std::{future::Future, sync::Arc};
+
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    sync::Notify,
+};
+
+pub trait AsciiStreamSocket {
+    fn send(&mut self, value: String) -> impl Future<Output = anyhow::Result<()>> + Send;
+    fn close(&mut self) -> impl Future<Output = ()> + Send;
+}
+
+#[tracing::instrument(skip_all)]
+pub async fn ascii_stream(
+    mut websocket: impl AsciiStreamSocket,
+    input_stream: impl tokio::io::AsyncRead + Unpin, // A file usually
+    shutdown_signal: Arc<Notify>,
+    when_new_chunk_appended: impl Fn() -> tokio::sync::oneshot::Receiver<()>,
+) -> anyhow::Result<()> {
+    info!("Starting ASCII streaming");
+    // write all the data from the input stream to the output stream
```

Comment:
style: Try to follow the style defined in STYLE.md.
```suggestion
    // Write all the data from the input stream to the output stream.
```